src/goose.rs

```diff
@@ -1513,62 +1657,67 @@ impl GooseUser {
 
     /// Tracks the time it takes for the current GooseUser to loop through all GooseTasks
     /// if Coordinated Omission Mitigation is enabled.
-    pub(crate) async fn update_request_cadence(&self, thread_number: usize) {
+    pub(crate) async fn update_request_cadence(&mut self, thread_number: usize) {
         if let Some(co_mitigation) = self.config.co_mitigation.as_ref() {
             // Return immediately if coordinated omission mitigation is disabled.
             if co_mitigation == &GooseCoordinatedOmissionMitigation::Disabled {
                 return;
             }
 
-            // Grab a read/write-copy of the `requests_cadence` object to update the
-            // current request cadence.
-            let mut request_cadence = self.request_cadence.write().await;
-
             // Grab the current timestamp to calculate the difference since the last
             // time through the loop.
             let now = std::time::Instant::now();
 
             // Swap out the `slept` counter, which is the total time the GooseUser slept
             // between tasks, a potentially randomly changing value. Reset to 0 for the
             // next loop through all GooseTasks.
-            request_cadence.delays_since_last_time = self.slept.swap(0, Ordering::SeqCst);
+            self.request_cadence.delays_since_last_time = self.slept;
+            self.slept = 0;
```

Comment:
Why not continue using swap here?

Reply:
I change `slept` from `AtomicU64` to `u64`, it's not possible to use swap anymore. 
Will `self.request_cadence.delays_since_last_time` always be `0` when I reach this part of the code? If yes I could use `std::mem::swap`.
Am I missing something?

Reply:
Perhaps I'm missing something. `self.request_cadence.delays_since_last_time` is always set to `self.slept` (which wouldn't be `0`), and then `self.slept` is set to `0`.

Reply:
Oh oki, I think we should only use `std::men::swap` if we really want to swap the value. In this case we reset one of the value to 0 right after the swap. It would make the code a bit unclear IMHO